l10n_nl_bsn/models/res_partner.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2016-2018 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models, _
+
+_logger = logging.getLogger(__name__)
+try:
+    from stdnum.nl import bsn
+except ImportError:
+    _logger.debug('Cannot `import stdnum.nl.bsn`.')
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    bsn_number = fields.Char(
+        string='BSN',
+        groups='hr.group_hr_user'
+    )
+
+    @api.multi
+    @api.onchange('bsn_number')
+    def onchange_bsn_number(self):
+        warning = {}
+        for partner in self:
+            if partner.bsn_number:
+                # properly format the entered BSN
+                partner.bsn_number = bsn.format(partner.bsn_number)
+
+                # check is valid, otherwise display a warning
+                warning = partner._warn_bsn_invalid()
+
+                # search for another partner with the same BSN
+                args = [('bsn_number', '=', partner.bsn_number),
+                        ('name', '!=', partner.name)]
```

Comment:
Why did you add a check on the name within the search domain?

Reply:
Because you may have multiple records representing the same partner. For example, if 2 records of _PartnerA_ are present: you may want to accept both records of _PartnerA_  having the same `bsn_number`.

Reply:
See also https://github.com/OCA/l10n-netherlands/pull/43

Reply:
Ok. Thanks for your reply.